lib/matplotlib/colors.py

```diff
@@ -3097,8 +3097,23 @@ def rgb_to_hsv(arr):
         dtype=np.promote_types(arr.dtype, np.float32),  # Don't work on ints.
         ndmin=2,  # In case input was 1D.
     )
+
     out = np.zeros_like(arr)
     arr_max = arr.max(-1)
+    arr_min = arr.min(-1)
+    # Check if input is in the expected range
+    if arr_max > 1:
+        raise ValueError(
+            "Input array must be in the range [0, 1]. "
+            f"Found a maximum value of {arr_max}"
+        )
+
+    if arr_min < 0:
+        raise ValueError(
+            "Input array must be in the range [0, 1]. "
+            f"Found a minimum value of {arr_min}"
```

Comment:
```suggestion
    if arr.min() < 0:
        raise ValueError(
            "Input array must be in the range [0, 1]. "
            f"Found a minimum value of {arr.min()}"
```
Since we don't need the mins anywhere else, we can simplify the code and inline the min calculation.
Yes, we do it double for the exception, but only when that raises, and in all likelyhood then the program will halt anyway and the double calculation cost doesn't matter.